test/e2e/copy_from_image_test.go

```diff
@@ -186,43 +218,85 @@ func TestCopyAnImageFromARepoToATarThatDoesNotContainNonDistributableLayersButTh
+		env := BuildEnv(t)
+		imgpkg := Imgpkg{t, Logger{}, env.ImgpkgPath}
+		defer env.Cleanup()
+
+		// general setup
+		testDir := env.Assets.CreateTempFolder("image-to-tar")
+		tarFilePath := filepath.Join(testDir, "image.tar")
+
+		nonDistributableLayerDigest := env.ImageFactory.PushImageWithANonDistributableLayer(env.RelocationRepo)
+		repoToCopyName := env.RelocationRepo + "include-non-distributable"
+
+		// copy to tar
+		imgpkg.Run([]string{"copy", "-i", env.RelocationRepo, "--to-tar", tarFilePath, "--include-non-distributable"})
+
+		var stdOutWriter bytes.Buffer
+		imgpkg.RunWithOpts([]string{"copy", "--tar", tarFilePath, "--to-repo", repoToCopyName}, RunOpts{
+			StdoutWriter: &stdOutWriter,
+			StderrWriter: &stdOutWriter,
+		})
+
+		if !strings.Contains(stdOutWriter.String(), "Skipped layer due to it being non-distributable.") {
+			t.Fatalf("Expected warning message to user, specifying which layer was skipped. But found: %s", stdOutWriter.String())
+		}
+		digestOfNonDistributableLayer, err := name.NewDigest(repoToCopyName + "@" + nonDistributableLayerDigest)
+		if err != nil {
+			t.Fatalf("Unable to determine the digest of the non-distributable layer. Got: %v", err)
+		}
+
+		layer, err := remote.Layer(digestOfNonDistributableLayer, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+		if err != nil {
+			t.Fatalf("Unable to fetch the layer of the copied image: %v", err)
+		}
+
+		_, err = layer.Compressed()
+		if err == nil {
+			t.Fatalf("Expected non-distributable layer to NOT be copied into registry, however it was")
+		}
 	})
 
-	if !strings.Contains(stdOutWriter.String(), "Skipped layer due to it being non-distributable.") {
-		t.Fatalf("Expected warning message to user, specifying which layer was skipped. But found: %s", stdOutWriter.String())
-	}
-	digestOfNonDistributableLayer, err := name.NewDigest(repoToCopyName + "@" + nonDistributableLayerDigest)
-	if err != nil {
-		t.Fatalf("Unable to determine the digest of the non-distributable layer. Got: %v", err)
-	}
-
-	layer, err := remote.Layer(digestOfNonDistributableLayer, remote.WithAuthFromKeychain(authn.DefaultKeychain))
-	if err != nil {
-		t.Fatalf("Unable to fetch the layer of the copied image: %v", err)
-	}
-
-	_, err = layer.Compressed()
-	if err == nil {
-		t.Fatalf("Expected non-distributable layer to NOT be copied into registry, however it was")
-	}
+	t.Run("airgapped environment", func(t *testing.T) {
+		env := BuildEnv(t)
+		imgpkg := Imgpkg{t, Logger{}, env.ImgpkgPath}
+		defer env.Cleanup()
+
+		// general setup
+		testDir := env.Assets.CreateTempFolder("image-to-tar")
+		tarFilePath := filepath.Join(testDir, "image.tar")
+
+		nonDistributableLayerDigest := env.ImageFactory.PushImageWithANonDistributableLayer(env.RepoForAirgappedTesting)
+		repoToCopyName := env.RelocationRepo + "include-non-distributable"
+
+		// copy to tar
+		imgpkg.Run([]string{"copy", "-i", env.RepoForAirgappedTesting, "--to-tar", tarFilePath, "--include-non-distributable"})
```

Comment:
what behaviour are we trying to test by using RepoForAirgappedTesting vs RelocationRepo?

Reply:
We want to write an assertion that when copying from a tarball (that was copied from registry A) to a registry B, that registry A is not reached at all.

The above is something that we caught manually while working on a performance improvement story.

Having 2 separate tests for 'internet' and 'without internet' is useful for the `performance` story the team is working on. 

Which is that when copying a tarball (from registry A) to registry B,  and there *is* internet available, *and* the registries are the same, the layers will be mounted. (and if the `--include-non-distributable-layer` flag is provided,, but those layers are missing in the tarball, those layers are also mounted too! <-- we got this 'feature' for free).

The assertions will verify that when there is internet that the non distributable layers are mounted (even though the tarball didn't contain those layers)

When there isn't any internet we will verify that the layers were not copied and that the appropriate error message is emitted.